packages/core/cli.js

```diff
@@ -41,9 +41,7 @@ listeners.forEach(listener => process.removeListener("warning", listener));
 let options = { logger: console };
 
 const inputArguments = process.argv.slice(2);
-const userWantsGeneralHelp =
-  (inputArguments[0] === "help" || inputArguments[0] === "--help") &&
-  inputArguments.length === 1;
+const userWantsGeneralHelp = inputArguments.length && ['help', '--help'].includes(inputArguments[0]);
```

Comment:
[These two tests](https://github.com/trufflesuite/truffle/pull/2971/checks?check_run_id=594035026#step:6:950) are failing because this code now returns `true` and calls `command.displayGeneralHelp` when passing `help <validOrInvalidCommand>` to the cli, instead of logging the particular commands help menu or logging the command passed is invalid.